d4s2_api_v1/api.py

```diff
@@ -24,48 +25,44 @@ def build_accept_url(request, transfer_id, delivery_type):
     return accept_url
 
 
-def prevent_email_template_set_in_request(request):
-    if request.data.get('email_template_set'):
-        raise ValidationError(CANNOT_PASS_EMAIL_TEMPLATE_SET)
-
-
-def get_email_template_for_request(request):
-    """
-    Given a request lookup the email_template associated with the user of the request.
-    If the user has not email template setup raises exception.
-    :param request: Request
-    :return: EmailTemplateSet
-    """
-    try:
-        user_email_template_set = UserEmailTemplateSet.objects.get(user=request.user)
-        return user_email_template_set.email_template_set
-    except UserEmailTemplateSet.DoesNotExist:
-        raise ValidationError(EMAIL_TEMPLATES_NOT_SETUP_MSG)
-
-
-def populate_email_template_in_request(request):
-    """
-    Given a request lookup the email template set associated with the user of the request and
-    store the value as 'email_template_set' in request.data. If the request already contains a
-    'email_template_set' this will raise an exception.
-    :param request: Request
-    """
-    prevent_email_template_set_in_request(request)
-    email_template_set = get_email_template_for_request(request)
-    request.data['email_template_set'] = email_template_set.id
-
-
-def prevent_null_email_template_set(obj):
-    if not obj.email_template_set:
-        raise ValidationError(ITEM_EMAIL_TEMPLATES_NOT_SETUP_MSG)
-
-
 class AlreadyNotifiedException(APIException):
     status_code = status.HTTP_400_BAD_REQUEST
     default_detail = 'Already notified'
 
 
-class DeliveryViewSet(viewsets.ModelViewSet):
+class ModelWithEmailTemplateSetMixin(object):
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(email_template_set=self.get_email_template_for_request(request))
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+    @staticmethod
+    def get_email_template_for_request(request):
```

Comment:
Can you elaborate on why this is a static method? It seems like instance-level functionality. The instance will have a `self.request` property, so if it were an instance method, `request` doesn't need to be passed as an argument.

Unfortunately, I can't seem to find documentation about where exactly `self.request` gets assigned, but it seems to be pretty fundamental in django class-based views

¯\\_(ツ)_/¯

Reply:
I wanted to make the code really obvious what was going on and minimize surprise. When you see a method that receives a `requests` parameter it is rather surprising that `self.requests` is also the same value.  I was able to find a couple places in DRF where they use `self.request` but nothing that spells it out. It's definitely there in the code. I will try out switching to the instance version and we can see how that looks.

Reply:
It was a `@staticmethod` so it could be used in the DeliveryPreview. The DeliveryPreview api endpoint doesn't make use of the create method so mixing an object that adds a create method seemed odd just to use one of the methods. You can see how this looks as an instance method in the latest commit:
```
class DeliveryPreviewView(ModelWithEmailTemplateSetMixin, generics.CreateAPIView):
...
    def create(self, request, *args, **kwargs):
        email_template_set = self.get_email_template_for_request()

        serializer = self.get_serializer(data=request.data)
        serializer.is_valid(raise_exception=True)
...
```